### ISSUE TITLE
🔍 Implement multiplayer micro-stack

### ISSUE THREAD

**Issue by Terkwood** [Terkwood/BUGOUT#174]
# Rationale 

In the current multiplayer implementation, the first person to visit the site in a given period has to endure an impossibly long wait time (75 seconds+) while the bigger box with Kafka starts up. Having a tiny little implementation dependent on Redis Streams can solve this. 

RAM footprint of the total system will decrease by about 100x.

Our goal is to sunset the Kafka implementation. 

## Write the following 

- [x] micro-judge (#133)
- [x] micro-changelog (#175)
- [x] micro-game-lobby (#184) 
- [x] micro-color-chooser (#334) 
- [x] micro-history-provider (also required by #238, PR #331 ) 
- [x] micro-sync #333 
- [x] simplify backend  extension for gateway (also see #205) 
- [x] #368 fake the providle / jvm-startup interaction
- [x] #332 gateway needs to coordinate events using redis streams,... 
- [x]  potentially using shared  libs defining common types (#339) 

## Testing and verification 

- [x] #362 functional testing 
- [x] #352 check make move payload across all boundaries

## Found work
 
- [x] fix botlink serialization woes caused by this effort #405 
- [x] carefully check updated DC.yml against tiny 
- [x] fix gateway r2d2 extension error #401
- [x] fix botlink regression #400
- [x] fix read error in micro-color-chooser #365 
- [x]  check into session abandonment in game lobby #370 
- [x]  make sure colorschosen event propagates back to client #371 
- [x]  ensure game state created correctly and visible to judge #378
- [x] fix private game rejection message #391 

## Admin cleanup 

- [x] #374 clean up admin scripts
- [x] set gateway, micro-judge, micro-changelog,  botlink release versions

## Deferred work

We don't need the participation service ported immediately, as it was never actually finished for the old JVM stack.  Here's the ticket related to that work: #325 

We can delete Kafka and all related apps, once this ticket is complete. See #229 

## Memory usage tracking

With `micro-judge v0.1.1`

```sh
 docker stats --format "table {{.Name}}\t{{.CPUPerc}}\t{{.MemUsage}}"
```

```text
NAME                           CPU %               MEM USAGE / LIMIT          
bugout_reverse-proxy_1         0.00%               5.852MiB / 62.77GiB
bugout_micro-judge_1           0.00%               4.223MiB / 62.77GiB
bugout_micro-color-chooser_1   0.00%               4.156MiB / 62.77GiB
bugout_micro-changelog_1       0.00%               4.859MiB / 62.77GiB
bugout_micro-sync_1            0.00%               4.297MiB / 62.77GiB
bugout_gateway_1               0.00%               4.5MiB / 62.77GiB
bugout_micro-game-lobby_1      0.00%               4.543MiB / 62.77GiB
bugout_redis_1                 0.05%               9.043MiB / 62.77GiB
```
